File: common/retex/src/main/kotlin/com/himamis/retex/renderer/android/geom/Point2DA.java
package com.himamis.retex.renderer.android.geom;

import android.graphics.PointF;

import com.himamis.retex.renderer.share.platform.geom.Point2D;

public class Point2DA implements Point2D {

    private PointF point;

    public Point2DA(double x, double y) {
        point = new PointF((float) x, (float) y);
    }

    public double getX() {
        return point.x;
    }

    public void setX(double x) {
        point.x = (float) x;
    }

    public double getY() {
        return point.y;
    }

    public void setY(double y) {
        point.y = (float) y;
    }

}

File: modules/retex/src/androidMain/kotlin/com/himamis/retex/renderer/android/geom/Point2DA.kt
package com.himamis.retex.renderer.android.geom

import android.graphics.PointF
import com.himamis.retex.renderer.share.platform.geom.Point2D

class Point2DA(x: Double, y: Double) : Point2D {
    private val point: PointF

    init {
        point = PointF(x.toFloat(), y.toFloat())
    }

    override fun getX(): Double {
        return point.x.toDouble()
    }

    override fun getY(): Double {
        return point.y.toDouble()
    }

    override fun setX(x: Double) {
        point.x = x.toFloat()
    }

    override fun setY(y: Double) {
        point.y = y.toFloat()
    }
}
